Indexing/TermIndex.cpp

```diff
@@ -62,10 +48,9 @@ void SuperpositionSubtermIndex::handleClause(Clause* c, bool adding)
                                               : EqHelper::getSubtermIterator(lit,_ord);
     while (rsti.hasNext()) {
       auto tt = TypedTermList(rsti.next());
-      ((TermSubstitutionTree*)_is)->handle(tt, lit, c, adding);
+      ((TermSubstitutionTree*)&*_is)->handle(tt, lit, c, adding);
```

Comment:
Couldn't the `&*` be dropped?

Reply:
No because _is is a `std::unique_ptr<T>` and not a `T*`  So the `*_is` is of type `T&` and `&*_is` is of type `T*`. It looks a bit odd but u need that pattern quite some times with smart pointers.